### PR TITLE
keycloak_client add option to support client-x509 authentication

### PR DESCRIPTION
**Pull request by Boolman**
##### SUMMARY

keycloak_client is missing support for x509 client authentication,   
this PR updates `client_authenticator_type` with the choice `client-x509`

##### ISSUE TYPE
Feature Pull Request

##### COMPONENT NAME
modules/keycloak_client

##### ADDITIONAL INFORMATION
Below is steps to test the result with this PR applied.
```
cat ~/keycloak/certs/start.sh  
export KEYCLOAK_ADMIN=admin
export KEYCLOAK_ADMIN_PASSWORD=admin


cd /opt/keycloak
bin/kc.sh start-dev --https-certificate-key-file=/certs/certificate.key --https-certificate-file=/certs/certificate.crt --https-trust-store-file=/certs/truststore.jks --https-trust-store-password=password --https-client-auth=request --log-level=trace
```
Start Keycloak
`docker run --rm -it -p 8080:8080 -p 8443:8443 -v ~/keycloak/certs:/certs --entrypoint=/bin/bash keycloak/keycloak:25.0 bash /certs/start.sh`

Apply config
```
ansible-playbook ....
```
Verify 
```
$ curl --cacert ~/keycloak/certs/ca.crt --cert ~/keycloak/certs/certificate.crt --key ~/keycloak/certs/certificate.key https://127.0.0.1:8443/realms/test/protocol/openid-connect/token --data "client_id=test&username=testuser&password=testuser&grant_type=password" -k -v
```

### CODE REVIEW

File: changelogs/fragments/8973-keycloak_client-add-x509-auth.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - keycloak_client - Add ``client-x509`` choise to ``client_authenticator_type`` (https://github.com/ansible-collections/community.general/pull/8973).
```
Comment (felixfontein) [inline changelogs/fragments/8973-keycloak_client-add-x509-auth.yml:2]:
```suggestion
  - keycloak_client - add ``client-x509`` choice to ``client_authenticator_type`` (https://github.com/ansible-collections/community.general/pull/8973).
```

File: plugins/modules/keycloak_client.py [hunk 2]
```diff
@@ -108,13 +108,14 @@
 
     client_authenticator_type:
         description:
-            - How do clients authenticate with the auth server? Either V(client-secret) or
-              V(client-jwt) can be chosen. When using V(client-secret), the module parameter
-              O(secret) can set it, while for V(client-jwt), you can use the keys C(use.jwks.url),
+            - How do clients authenticate with the auth server? Either V(client-secret),
+              V(client-jwt) or V(client-x509) can be chosen. When using V(client-secret), the module parameter
```
Comment (felixfontein) [inline plugins/modules/keycloak_client.py:112]:
```suggestion
              V(client-jwt), or V(client-x509) can be chosen. When using V(client-secret), the module parameter
```
https://docs.ansible.com/ansible/devel/dev_guide/style_guide/grammar_punctuation.html#commas

File: plugins/modules/keycloak_client.py [hunk 3]
```diff
@@ -581,6 +582,14 @@
                     - For OpenID-Connect clients, client certificate for validating JWT issued by
                       client and signed by its key, base64-encoded.
 
+            x509.subjectdn:
+                description:
+                    - For OpenID-Connect clients, subject which will be used to authenticate the client.
```
Comment (felixfontein) [inline plugins/modules/keycloak_client.py:587]: This is a new suboption, so it needs to be announced as added in the next release. Also, you need to specify a type. Is this a string? Then use `type: str`.
```suggestion
                    - For OpenID-Connect clients, subject which will be used to authenticate the client.
                type: ...
                version_added: 9.5.0
```

File: plugins/modules/keycloak_client.py [hunk 4]
```diff
@@ -581,6 +582,14 @@
...
+
+            x509.allow.regex.pattern.comparison:
+                description:
+                    - For OpenID-Connect clients, boolean specifying whether to allow C(x509.subjectdn) as regular expression.
```
Comment (felixfontein) [inline plugins/modules/keycloak_client.py:591]: Same as above. Likely `type: bool` if this is a boolean.
```suggestion
                    - For OpenID-Connect clients, boolean specifying whether to allow C(x509.subjectdn) as regular expression.
                type: ...
                version_added: 9.5.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution. I've added some first comments below.

File: plugins/modules/keycloak_client.py (hunk 3 above)
Comment (Boolman) [reply plugins/modules/keycloak_client.py:587]: None of the other suboptions under `attributes` have a type specified, I assumed these was merely examples, is this not correct?

Comment (russoz) [reply plugins/modules/keycloak_client.py:587]: Nope, these are not examples, they are the source for the module documentation, which will be rendered to HTML and published later to the official documentation site, in the case of this module, here:

https://docs.ansible.com/ansible/latest/collections/community/general/keycloak_client_module.html#ansible-collections-community-general-keycloak-client-module

So, whilst other suboptions might not specify types, in general the more information we can include here, the more complete our user documentation is going to be.

**Review by russoz — COMMENTED** [review]
Hi @Boolman  thanks for your contribution!

Just left a small comment below, about the docs.

Other than that, I am assuming (lazily not checking the code) that the simple fact of adding new keys to `attributes` makes them be passed through KC's API. If that's not the case, then maybe more code is needed to ensure the new options are passed along.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@Boolman thanks for your contribution!
@russoz thanks for reviewing!

**Comment by felixfontein** [comment]
FYI @Boolman you might be interested in https://forum.ansible.com/t/keycloak-modules-in-community-general/41746

**Pull request merged**